src/Compilers/VisualBasic/Portable/Semantics/OverloadResolution.vb

```diff
@@ -3404,11 +3404,15 @@ Bailout:
                 Dim passedExpression As BoundExpression = binder.PassArgumentByVal(argument, conv, targetType, BindingDiagnosticBag.Discarded)
 
                 If Not passedExpression.IsConstant Then ' Trying to match native compiler behavior in Semantics::IsValidAttributeConstant
-                    Dim visitor As New Binder.AttributeExpressionVisitor(binder, passedExpression.HasErrors)
-                    visitor.VisitExpression(passedExpression, BindingDiagnosticBag.Discarded)
-
-                    If visitor.HasErrors Then
+                    If Not targetType.IsValidTypeForAttributeArgument(binder.Compilation) Then
```

Comment:
>Not targetType.IsValidTypeForAttributeArgument(binder.Compilation) [](http://example.com/codeflow?start=23&length=66)

It looks like there is a t least one more place where ```AttributeExpressionVisitor``` is used and that place is not "protected". I think fixing the visitor is a better approach.